cpp/include/cudf/scalar/scalar_device_view.cuh

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/strings/string_view.cuh>
+#include <cudf/types.hpp>
+
+namespace cudf {
+
+namespace detail {
+
+/**---------------------------------------------------------------------------*
+ * @brief An immutable, non-owning view of scalar from device that is trivially 
+ * copyable and usable in CUDA device code.
+ *---------------------------------------------------------------------------**/
+class alignas(16) scalar_device_view_base {
```

Comment:
Suggest explaining the reason for `alignas(16)` in a note in the class docs.

Reply:
I was going to comment that we don't actually need the `alignas` here because I don't know of a situation where we're going to pass an array of `scalar_device_view` objects into device code.